### PR TITLE
add CORS to http endpoint

### PR DESCRIPTION
**Pull request by jbtrystram**
I am not sure what origin (if any) we should allow by default here ?

### CODE REVIEW

File: http-endpoint/src/lib.rs [hunk 1]
```diff
@@ -44,6 +45,13 @@ pub struct Config {
 
     #[serde(default)]
     pub http: HttpConfig,
+
```
Comment (ctron) [inline http-endpoint/src/lib.rs:48]: I think we should extract that into a dedicated struct, having a field `cors` in the http configuration.

File: http-endpoint/src/lib.rs [hunk 2]
```diff
@@ -107,27 +115,46 @@ pub async fn run(config: Config, startup: &mut dyn Startup) -> anyhow::Result<()
     }
 
     let main = HttpBuilder::new(config.http, Some(startup.runtime_config()), move |cfg| {
+        let mut cors = Cors::default()
```
Comment (ctron) [inline http-endpoint/src/lib.rs:118]: So you could to an `let cors: Cors = config.http.cors.into();'`

Comment (ctron) [reply http-endpoint/src/lib.rs:118]: Yes, the `into` needs to provide a few additional settings, like the code below that comment.

**Comment by jbtrystram** [comment]
This depends on https://github.com/drogue-iot/drogue-bazaar/pull/3

**Comment by ctron** [comment]
I think you need to rebase the PR

**Pull request merged**